### DISCUSSION TITLE
WebOS build

### DISCUSSION THREAD

**Discussion by ThaDaVos** [lamarios/clipious#298] [General]
Instead of continuing in the issue - I think it's better to continue here:
https://github.com/lamarios/clipious/issues/296

Quick question, is the web version buildable? I tried and getting all sorts of errors - it seems it's trying to load non-web dependencies too

**Comment by lamarios** [comment]
for now only android is supported. one of the big  reasons why web is not supported is the DB i used (objectbox). it's possible to make an interface of the DB (db.dart) and have two different ways of handling the DB. One for platforms that support objectbox and one for those who don't

**Reply by ThaDaVos** [reply]
Yeah, figured it was that - already looking into it - gotta learn a little dart

Would you be okay if I try to refactor it and push a PR for that?

The WebOs build really only needs the web build to function and then I can wrap it and start working on the integration - but offering a web version of your amazing UI would be great also!

**Reply by ThaDaVos** [reply]
Gonna use this and interfaces/abstracts:
https://dart.dev/guides/libraries/create-packages#conditionally-importing-and-exporting-library-files

**Reply by lamarios** [reply]
yeah I was going to do it for unit test, can check the existing unit test PR

**Reply by ThaDaVos** [reply]
Ah I see - as it also helps with mocking during tests

**Reply by ThaDaVos** [reply]
I also came across this - the benchmark though:
https://github.com/isar/isar

**Reply by ThaDaVos** [reply]
Something I preferably use for projects with databases is the repository design pattern or facade/strategy design pattern for the above mentioned reasons:
https://refactoring.guru/design-patterns/facade
https://refactoring.guru/design-patterns/strategy

@lamarios - do you want me to continue from your PR to build out the interfaces/abstracts or should I start clean?

**Reply by lamarios** [reply]
Let's keep it simple, just create an interface and implement. 

I think that PR is too old and outdated as i refactored the whole code base since then. It will take more time for you to fix conflicts than starting anew.

**Reply by ThaDaVos** [reply]
Yeah I was thinking the same - gonna make a interface from `DBClient` and create two concretes - one for ObjectBox and one for something which is Web compatible
